File: src/main/java/io/transwarp/logparser/filter/ExceptionFilter.java
package io.transwarp.logparser.filter;

import io.transwarp.logparser.util.Record;

/**
 * Author: stk
 * Date: 2018/4/2
 */
public class ExceptionFilter implements Filter {
    @Override
    public boolean filter(Record record) {
        return record.isException();
    }

    @Override
    public void clean() {
    }
}

File: src/main/scala/io/transwarp/logparser/filter/ExceptionFilter.scala
package io.transwarp.logparser.filter

import io.transwarp.logparser.util.LogEntry

/**
  * Author: stk
  * Date: 2018/4/3
  */
class ExceptionFilter(override val params: List[String]) extends Filter {
  override def filter(logEntry: LogEntry): Boolean = Some(logEntry).exists(_.isException)

  override def copy(): Filter = new ExceptionFilter(params)
}
